src/_gettsim/taxes/lohnst.py

```diff
@@ -0,0 +1,401 @@
+from math import trunc
+
+from _gettsim.shared import add_rounding_spec
+from _gettsim.taxes.eink_st import _eink_st_tarif
+
+
+@add_rounding_spec(params_key="lohnst")
+def lohnst_eink(
+    bruttolohn_m: float,
+    steuerklasse: int,
+    eink_st_abzuege_params: dict,
+    vorsorgepauschale: float,
+) -> float:
+    """Calculate tax base for Lohnsteuer (withholding tax on earnings).
+
+    Parameters
+    ----------
+    bruttolohn_m:
+      See basic input variable :ref:`bruttolohn_m <bruttolohn_m>`.
+    steuerklasse:
+      See :func:`steuerklasse`
+    eink_st_abzuege_params:
+      See :func:`eink_st_abzuege_params`
+    vorsorgepauschale
+        See :func:`vorsorgepauschale`
+
+    Returns
+    -------
+
+    """
+    entlastung_freibetrag_alleinerz = (steuerklasse == 2) * eink_st_abzuege_params[
+        "alleinerz_freibetrag"
+    ]
+
+    if steuerklasse == 6:
+        werbungskosten = 0
+    else:
+        werbungskosten = eink_st_abzuege_params["werbungskostenpauschale"]
+
+    if steuerklasse == 6:
+        sonderausgaben = 0
+    else:
+        sonderausgaben = eink_st_abzuege_params["sonderausgabenpauschbetrag"]["single"]
+
+    # Zu versteuerndes Einkommen / tax base for Lohnsteuer.
+    out = max(
+        12 * bruttolohn_m
+        - werbungskosten
+        - sonderausgaben
+        - entlastung_freibetrag_alleinerz
+        - vorsorgepauschale,
+        0.0,
+    )
+
+    return out
+
+
+def _lohnsteuer_klasse5_6_basis(taxable_inc: float, eink_st_params: dict) -> float:
+    """Calculate base for Lst for Steuerklasse 5 and 6.
+
+    §39 b Absatz 2 Satz 7 (part 1):
+
+        Jahreslohnsteuer die sich aus dem Zweifachen des Unterschiedsbetrags zwischen
+        dem Steuerbetrag für das Eineinviertelfache und dem Steuerbetrag für das
+        Dreiviertelfache des zu versteuernden Jahresbetrags nach § 32a Absatz 1 ergibt;
+        die Jahreslohnsteuer beträgt jedoch mindestens 14 Prozent des zu versteuernden
+        Jahresbetrags.
+
+    Parameters
+    ----------
+
+    taxable_inc:
+        Taxable Income used in function (not necessarily the same as lohnst_eink)
+    eink_st_params
+        See params documentation :ref:`eink_st_params <eink_st_params>`
+
+    Returns
+    -------
+    Base for Lohnsteuer for Steuerklasse 5 and 6
+
+    """
+
+    out = max(
+        2
+        * (
+            trunc(_eink_st_tarif(taxable_inc * 1.25, eink_st_params))
```

Comment:
Please work with `@add_rounding_spec` instead of truncating here, if rounding is turned off, we want to avoid discontinuities everywhere.

Reply:
@JakobWegmann : I removed the `trunc` statements in the steuerklasse 5/6 calculation. Tests still work.